### PR TITLE
Mypy checks + fixes

### PR DESCRIPTION
**Pull request by MissingNO57**
- Additional linting checks (MyPy)

- removed 
      ```repo_root_path = Path(__file__).parent.parent.parent.absolute()
sys.path.insert(0, str(repo_root_path))```

---

## Code Review Checklist (to be filled out by reviewer)

- [x] Description accurately reflects what changes are being made.
- [x] Either the PR references an issue (via the "Development" combobox) or the description explains the need for the changes.
- [x] The PR appropriately sized.
- [x] The PR contains an idempotent DB migration.
- [x] I have verified the correctness of the DB migration using relevant data (e.g. test-generated data).
- [x] New code has enough tests.
- [x] New code has enough documentation to answer "how do I use it?" and "what does it do?".
- [x] Existing documentation is up-to-date, if impacted.

### CODE REVIEW

File: src/genesis/helpers/field_enums.py [hunk 1]
```diff
@@ -86,9 +89,8 @@ class EventFields(NamedFields):
     block_id = 2
     type = 3
 
-    @classmethod
```
Comment (MissingNO57) [inline src/genesis/helpers/field_enums.py:89]: removal of `classmethod`s for:
`error: Only instance methods can be decorated with @property  [misc]`

File: src/genesis/helpers/field_enums.py [hunk 2]
```diff
@@ -32,10 +33,15 @@ def select_query(cls, tables: List[str] = None, prefix=False) -> str:
...
     def select_where(
-        cls, where_clause: str, tables: List[str] = None, prefix=False
+        cls, where_clause: str, tables: Optional[List[str]] = None, prefix=False
     ) -> str:
         return f"{cls.select_query(tables=tables, prefix=True)} WHERE {where_clause}"
 
+    @classmethod
+    @abstractmethod
+    def get_table(cls):
+        pass
```
Comment (MissingNO57) [inline src/genesis/helpers/field_enums.py:43]: Combination of `@abstractmethod`, `@classmethod` and `@property` is problematic
I haven't found other solution than just transform it into abstract class method.

File: src/genesis/observers/chain_id.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ def subscribe_to(
     ) -> DisposableBase:
         operators = [
             filter_(self.filter_chain_id),
-            map_(lambda next_: next_[1]),
+            map_(lambda next_: next_[1]),  # type: ignore
```
Comment (MissingNO57) [inline src/genesis/observers/chain_id.py:26]: next_ is not Iterable

File: src/genesis/observers/chain_id.py [hunk 4]
```diff
@@ -15,15 +15,20 @@ class ChainIdObserver(Observer):
...
     def __init__(self, on_next=None, on_completed=None, on_error=None) -> None:
         super().__init__(on_next=on_next, on_completed=on_completed, on_error=on_error)
 
     def subscribe_to(
         self, observable: Observable, scheduler: Optional[Scheduler] = None
     ) -> DisposableBase:
+
         operators = [
             filter_(self.filter_chain_id),
-            map_(lambda next_: next_[1]),
```
Comment (MissingNO57) [inline src/genesis/observers/chain_id.py:26]: Mypy had issues with resoving lambda

One option was to move lambda to separate line
```
map_chain_id: Callable[Tuple[str, Any], str] = lambda next_: next_[1]
```

But I prefer this as a cleaner solution.

File: tests/helpers/contracts.py [hunk 5]
```diff
@@ -80,7 +80,7 @@ def _instantiate(self) -> Address:
 
 class BridgeContract(LedgerContract):
     admin: Wallet = None
-    cfg: BridgeContractConfig = None
+    cfg: BridgeContractConfig
```
Comment (MissingNO57) [inline tests/helpers/contracts.py:83]: BridgeContractConfig cannot be None as it's not Optional[BridgeContractConfig]. Making it Optional would require additional ifs and assertions.

File: tests/helpers/contracts.py [hunk 6]
```diff
@@ -106,4 +106,4 @@ def _store(self) -> int:
 
     def _instantiate(self) -> Address:
         assert (self.admin and self.cfg) is not None
-        return self.instantiate(self.cfg.to_dict(), self.admin)
+        return self.instantiate(self.cfg.to_dict(), self.admin)  # type: ignore
```
Comment (MissingNO57) [inline tests/helpers/contracts.py:109]:
```
tests/helpers/contracts.py:109: error: "BridgeContractConfig" has no attribute "to_dict"
```

Looks like MyPy doesn't know how to resolve
```
@dataclass_json
@dataclass
```

File: src/genesis/state/bank.py [hunk 7]
```diff
@@ -25,7 +25,7 @@ class BankStateData:
 
 
 class BankState(OwnAttrsMixin, BankStateData):
-    def __init__(self, **kwargs: Dict[str, any]):
```
Comment (MissingNO57) [inline src/genesis/state/bank.py:28]: Super type __init__ was complaining about the datatype of kwargs

File: src/genesis/state/utils.py [hunk 8]
```diff
@@ -18,7 +18,7 @@ def attrs(self) -> List[str]:
 
 class ListConstructorMixin:
     @classmethod
-    def from_dict_list(cls, list_: Union[List[Dict], type(None)]):
+    def from_dict_list(cls, list_: Optional[Union[List[Dict], Dict]]):
```
Comment (bryanchriswhite) [inline src/genesis/state/utils.py:21]:
```suggestion
    def from_dict_list(cls, list_: Optional[List[Dict]]):
```

**Review by bryanchriswhite — APPROVED** [review]
Thanks @MissingNO57! :raised_hands:

**Pull request merged**